app/uk/gov/hmrc/transitmovements/repositories/DeparturesRepository.scala

```diff
@@ -53,15 +55,15 @@ class DeparturesRepositoryImpl @Inject() (
     with DeparturesRepository
     with Logging {
 
-  def insert(departure: Departure): EitherT[Future, Throwable, DeclarationResponse] =
+  def insert(departure: Departure): EitherT[Future, MongoError, DeclarationResponse] =
     EitherT {
       retry(
         attempts = 0,
         attempt = () => insertDocument(departure)
       )
     }
 
-  private def insertDocument(departure: Departure): Future[Either[Throwable, DeclarationResponse]] =
+  private def insertDocument(departure: Departure): Future[Either[MongoError, DeclarationResponse]] =
```

Comment:
this name seems a bit off, since you don't provide a document, you provide a departure. Maybe tweak the name?

Another consideration is whether this should be a private method or an internal method - are we intending to reuse this code outside of insert?